tenant/src/main/scala/com/tenant/TenantServer.scala

```diff
@@ -0,0 +1,52 @@
+package com.tenant
+
+import akka.actor.typed.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.{HttpRequest, HttpResponse}
+import com.typesafe.config.ConfigFactory
+
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+object TenantServer {
+  def main(args: Array[String]): Unit = {
+    println("running main of TenantServer")
+    // Important: enable HTTP/2 in ActorSystem's config
+    // We do it here programmatically, but you can also set it in the application.conf
+    val conf = ConfigFactory.parseString("akka.http.server.preview.enable-http2 = on")
+      .withFallback(ConfigFactory.defaultApplication())
+    val system = ActorSystem[TestActor.TestCommand](TestActor(), "TenantServer", conf)
+    new TenantServer(system).run()
+    // ActorSystem threads will keep the app alive until `system.terminate()` is called
+  }
+}
+
+class TenantServer(system: ActorSystem[TestActor.TestCommand]) {
+  def run(): Future[Http.ServerBinding] = {
+    // Akka boot up code
+    implicit val sys = system
+    implicit val ec: ExecutionContext = system.executionContext
+
+    // Create service handlers
+    val service: HttpRequest => Future[HttpResponse] =
+      TenantServiceHandler.withServerReflection(new TenantServiceImpl(system))
+
+    val bound: Future[Http.ServerBinding] = Http(system)
+      .newServerAt(interface = "0.0.0.0", port = 8080)
```

Comment:
Used 0.0.0.0 instead of 127.0.0.1 because using localhost for some reason makes Docker listen with IPv4 which causes issues. Using 0.0.0.0 makes docker listen with IPv6 which allows the gRPC server to properly work.